src/test/java/com/example/javaee_jpa_hibernate/repository/InvoiceRepositoryTest.java

```diff
@@ -0,0 +1,170 @@
+package com.example.javaee_jpa_hibernate.repository;
+
+import com.example.javaee_jpa_hibernate.model.Invoice;
+import com.example.javaee_jpa_hibernate.model.InvoiceBody;
+import com.example.javaee_jpa_hibernate.model.counterparty.Counterparty;
+import com.example.javaee_jpa_hibernate.model.invoice_item.InvoiceItem;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class InvoiceRepositoryTest {
+
+    @Test
+    public void shouldCreateAnInvoice() throws Exception {
+        //given
+        EntityManagerFactory entityManagerFactory = Persistence
```

Comment:
This should be shared by all test cases and extracted as a field. You can create this object in a method annotated as `@BeforeAll`

Reply:
done